### PR TITLE
Add dropbox support, fix LAN locking issue, make MonUtil support removab...

### PR DESCRIPTION
**Pull request by Mike-GC**
The main point of this change is to add support for primitive cloud support by syncing over dropbox (or onedrive, etc.).

Along the way, we had to make settings engine monitor all remotes again (not just network remotes) - this is necessary to add dropbox support, but in order not to regress the USB drive scenario, required MonUtil to support monitoring removable drives (and still allow them to be removed).

We also get rid of the silly "last real change" file next to remotes, which was sort of a hack around SQL CE's stupid behavior (it would always modify the database even if you open it just for reading), and which was not a valid solution for the dropbox scenario. Now we copy the DB locally, work on it, then re-upload to temp file in remote directory, then move back to original remote database path as an atomic operation.

A side effect of that "copy locally" part of the change is that now a client syncing to a network remote then loses its connection while doing so, no longer leaves the main database locked (which previously preventing all clients from syncing).

tl;dr - we can now sync over dropbox, and fix a couple of bugs along the way.

### CODE REVIEW

File: src/SettingsEngine/browse/window.cpp [hunk 1]
```diff
@@ -1487,13 +1487,19 @@ LRESULT CALLBACK BrowseWindow::WndProc(
         fCsEntered = TRUE;
         if (SUCCEEDED(UXDATABASE(dwIndex).hrSyncResult) && NULL != UXDATABASE(dwIndex).pcplConflictProductList)
         {
-            hr = TrayShowBalloon(L"Sync Conflicts", L"Sync conflicts occurred. Click here to open main browser window, and resolve them.", NIIF_WARNING);
-            ExitOnFailure(hr, "Failed to show tray message");
+            hrTemp = TrayShowBalloon(L"Sync Conflicts", L"Sync conflicts occurred. Click here to open main browser window, and resolve them.", NIIF_WARNING);
+            if (FAILED(hrTemp))
+            {
+                LogStringLine(REPORT_STANDARD, "Failed to show tray balloon");
```
Comment (robmen) [inline src/SettingsEngine/browse/window.cpp:1493]: You'll want hrTemp in here one day.

File: src/SettingsEngine/browse/window.cpp [hunk 2]
```diff
@@ -1487,13 +1487,19 @@ LRESULT CALLBACK BrowseWindow::WndProc(
...
+            }
         }
         else
         {
-            hr = TrayHideBalloon();
-            ExitOnFailure(hr, "Failed to hide tray message");
+            hrTemp = TrayHideBalloon();
+            if (FAILED(hrTemp))
+            {
+                LogStringLine(REPORT_STANDARD, "Failed to hide tray balloon");
```
Comment (robmen) [inline src/SettingsEngine/browse/window.cpp:1501]: Ditto.
